staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere.go

```diff
@@ -1519,6 +1531,51 @@ func (vs *VSphere) NodeAdded(obj interface{}) {
 	if err := vs.nodeManager.RegisterNode(node); err != nil {
 		klog.Errorf("failed to add node %+v: %v", node, err)
 	}
+	vs.setNodeZoneLabels(node)
```

Comment:
This appears to be the only place where the labels are set, and it is called only once for each node object. Wouldn't it result in nodes without zone labels at all in a cluster where the labels should be set?

Reply:
Shouldn't. This thing calls when kcm recognizes new node and adds it to internal nodeManager structure.